```
In file included from Python/ceval.c:770:
Python/generated_cases.c.h:3437:48: warning: code will never be executed [-Wunreachable-code]
            if (0) { stack_pointer[-(1 + 0)] = res2; }
                                               ^~~~
Python/generated_cases.c.h:3437:17: note: silence by adding parentheses to mark code as explicitly dead
            if (0) { stack_pointer[-(1 + 0)] = res2; }
                ^
                /* DISABLES CODE */ ( )
Python/generated_cases.c.h:3415:48: warning: code will never be executed [-Wunreachable-code]
            if (0) { stack_pointer[-(1 + 0)] = res2; }
                                               ^~~~
Python/generated_cases.c.h:3415:17: note: silence by adding parentheses to mark code as explicitly dead
            if (0) { stack_pointer[-(1 + 0)] = res2; }
                ^
                /* DISABLES CODE */ ( )
clang -c -fno-strict-overflow -Wsign-compa
```

Warning logs are reduced, but not sure this is a safe way.

--- a/Python/generated_cases.c.h
+++ b/Python/generated_cases.c.h
@@ -3412,7 +3412,7 @@
             res = Py_NewRef(descr);
             STACK_GROW(0);
             stack_pointer[-1] = res;
-            if (0) { stack_pointer[-(1 + 0)] = res2; }
+
             next_instr += 9;
             DISPATCH();
         }
@@ -3434,7 +3434,7 @@
             res = Py_NewRef(descr);
             STACK_GROW(0);
             stack_pointer[-1] = res;
-            if (0) { stack_pointer[-(1 + 0)] = res2; }
+
             next_instr += 9;
             DISPATCH();
         }

--- a/Python/generated_cases.c.h
+++ b/Python/generated_cases.c.h
@@ -3412,7 +3412,6 @@
             res = Py_NewRef(descr);
             STACK_GROW(0);
             stack_pointer[-1] = res;
-
             next_instr += 9;
             DISPATCH();
         }
@@ -3434,7 +3433,6 @@
             res = Py_NewRef(descr);
             STACK_GROW(0);
             stack_pointer[-1] = res;
-
             next_instr += 9;
             DISPATCH();
         }

--- a/Python/generated_cases.c.h
+++ b/Python/generated_cases.c.h
@@ -3391,7 +3391,6 @@
 
         TARGET(LOAD_ATTR_NONDESCRIPTOR_WITH_VALUES) {
             PyObject *self = stack_pointer[-1];
-            PyObject *res2 = NULL;
             PyObject *res;
             uint32_t type_version = read_u32(&next_instr[1].cache);
             uint32_t keys_version = read_u32(&next_instr[3].cache);
@@ -3418,7 +3417,6 @@
 
         TARGET(LOAD_ATTR_NONDESCRIPTOR_NO_DICT) {
             PyObject *self = stack_pointer[-1];
-            PyObject *res2 = NULL;
             PyObject *res;
             uint32_t type_version = read_u32(&next_instr[1].cache);
             PyObject *descr = read_obj(&next_instr[5].cache);

--- a/Tools/cases_generator/generate_cases.py
+++ b/Tools/cases_generator/generate_cases.py
@@ -226,7 +226,7 @@ class Formatter:
             self.emit(f"STACK_GROW({osym});")
 
     def declare(self, dst: StackEffect, src: StackEffect | None):
-        if dst.name == UNUSED:
+        if dst.name == UNUSED or dst.cond == "0":
             return
         typ = f"{dst.type}" if dst.type else "PyObject *"
         if src:

--- a/Tools/cases_generator/generate_cases.py
+++ b/Tools/cases_generator/generate_cases.py
@@ -251,7 +251,10 @@ class Formatter:
         else:
             stmt = f"{dst.name} = {cast}{src.name};"
             if src.cond:
-                stmt = f"if ({src.cond}) {{ {stmt} }}"
+                if src.cond == "0":
+                    stmt = "" # skip
+                else:
+                    stmt = f"if ({src.cond}) {{ {stmt} }}"
             self.emit(stmt)
 
     def cast(self, dst: StackEffect, src: StackEffect) -> str:

--- a/Tools/cases_generator/generate_cases.py
+++ b/Tools/cases_generator/generate_cases.py
@@ -252,9 +252,9 @@ class Formatter:
             stmt = f"{dst.name} = {cast}{src.name};"
             if src.cond:
                 if src.cond == "0":
-                    stmt = "" # skip
-                else:
-                    stmt = f"if ({src.cond}) {{ {stmt} }}"
+                    # It will not be executed
+                    return
+                stmt = f"if ({src.cond}) {{ {stmt} }}"
             self.emit(stmt)
 
     def cast(self, dst: StackEffect, src: StackEffect) -> str:

